### PR TITLE
Refactor FXIOS-6932 [v120] Additional wiring for new Sync tab and table cells

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6932)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15411)

## :bulb: Description

This adds support in the new redux-based Sync tab for configuring the client headers and tab cells using data from the Redux state (which has a property which continues to use `ClientAndTabs`). For now it's reusing the existing header views and cells, per team feedback, since there doesn't seem to be any need to duplicate those, and they're used in several places in the app. 

Also adds wiring to support collapsing the section headers to match previous functionality and some initial unit tests for the new remote panel.

Note: aspects of the new sync tab are still a work-in-progress, additional updates forthcoming.

## 🖼️ Screenshot

test

Screenshot 2023-10-04 at 5 02 06 PM

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Browser/Tabs/TabTrayViewController.swift [hunk 1]
```diff
@@ -435,6 +436,7 @@ class TabTrayViewController: UIViewController,
 
     func remotePanel(didSelectURL url: URL, visitType: VisitType) {
         TelemetryWrapper.recordEvent(category: .action, method: .open, object: .syncTab)
+        // TODO: Need to provide handler for didSelectURL. Forthcoming.
```
Comment (OrlaM) [inline Client/Frontend/Browser/Tabs/TabTrayViewController.swift:439]: nit: TODO's should have a ticket number

File: Client/Frontend/Browser/Tabs/TabTrayViewController.swift [hunk 2]
```diff
@@ -263,7 +263,8 @@ class TabTrayViewController: UIViewController,
         case .privateTabs:
             return TabDisplayViewController(isPrivateMode: true)
         case .syncedTabs:
-            let panel = RemoteTabsPanel(state: RemoteTabsPanelState.emptyState())
+            let state = RemoteTabsPanelState.emptyState() // Temporary
```
Comment (OrlaM) [inline Client/Frontend/Browser/Tabs/TabTrayViewController.swift:266]: Similarly, for the temporary comment can you add a TODO with the ticket number it's to be updated with.

**Review by OrlaM — APPROVED** [review]
A couple of small nits but otherwise looks good.

File: Client/Frontend/Browser/Tabs/RemoteTabs/RemoteTabsTableViewController.swift [hunk 3]
```diff
@@ -215,31 +221,99 @@ class RemoteTabsTableViewController: UITableViewController,
...
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        if isShowingEmptyView {
-            return 0
-        } else {
-            // TODO: Show clients and tabs. Forthcoming. [FXIOS-6942]
-            return 0
-        }
+        guard !isShowingEmptyView else { return 0 }
+
+        guard !hiddenSections.contains(section) else { return 0 }
```
[the remark this answers is not in the document]
Comment (mattreaganmozilla) [reply Client/Frontend/Browser/Tabs/RemoteTabs/RemoteTabsTableViewController.swift]: ✅

File: Client/Frontend/Browser/Tabs/TabTrayViewController.swift (hunk 1 above)
Comment (mattreaganmozilla) [reply Client/Frontend/Browser/Tabs/TabTrayViewController.swift]: ✅ Added ticket numbers; I'm not 100% sure which epic subtasks will cover some of this work but tried to pick the one that looked most appropriate.

**Comment by mattreaganmozilla** [comment]
Going to merge but if there is any additional feedback just LMK and happy to put up a follow-up PR. Thank you (cc @yoanarios )

**Pull request merged**